### DISCUSSION TITLE
Audio doesn't get sent to proper output

### DISCUSSION THREAD

**Discussion by Ethereal-Dev** [ppy/osu#15962] [Q&A] [answered]
Hello everyone,

For the past ~2 weeks I've been playing osu! with a virtual mixer (voicemeeter potato) and setting osu! to output the game sound directly to my headphones to bypass the virtual mixer to avoid any delay.

**My issue**
This has been working like a charm for around two weeks now, but 30 minutes ago this stopped working while I was playing.
Now, when I attempt to route my game audio directly to my headphones using the in-game output device setting it routes the sound to the virtual mixer anyway. When I attempt to route my game sound to my monitor instead, it works as expected and the game sound comes through my monitor.

**My attempts to fix it**
(None of the following things have resolved the issue)
I have tried:
1. Re-installing osu!
2. Re-installing voicemeeter potato (the virtual mixer)
3. Changing the windows default output device and using the osu! default output device option

**What does work**
I have discovered through some research that after disabling the virtual mixer I can once again route my game sound to my headphones without issue, but then I do not have desktop audio anymore, and if I attempt to re-enable the virtual mixer then the issue returns, leaving me without game sound.

This way, I can at least keep playing osu!, but this is not a "real" fix.

Short video using OBS to show where the audio is going.
the "osu audio" track is my speaker output
the "desktop audio" is my virtual mixer
https://youtu.be/5T73efMbEt4

Does anyone have any idea how I can resolve this issue?

(Posted this in Q&A since I'm not 100% sure if this is an osu! issue or not.)

**Comment by bdach** [comment]
the video attached shows stable. this repository is accepting reports and discussions concerning lazer only.

**Comment by Ethereal-Dev** [comment, marked as answer]
Alright, where should i put this instead?

**Reply by integerrr** [reply]
https://github.com/ppy/osu-stable-issues

**Reply by Ethereal-Dev** [reply]
Alright, thanks, will post this there.

**Reply by bdach** [reply]
note that you should probably report that to the above link *if* you are sure this is a game issue. which this may well not be.

i'd recommend asking around in places related to the virtual mixer software first since i'd wager it's more likely the thing at fault here. failing that, it's 99% likely you're just going to have to figure it out on your own.

**Reply by Ethereal-Dev** [reply]
I'm pretty confident in saying that this _is_ probably an osu! issue. The virtual mixer does not redirect all PC audio, so applications like osu! should be able to send their audio where ever the user sets it.

Though you are probably right about me having to fix this myself, no one else seems to have ever had this issue, or I can't find any information on this at least.
